Add .settings and .project from Eclipse to gitignore

sergeichestakov opened the pull request:
# Why

.settings and .project from Eclipse, or the Buildship plugin for Gradle, are not being ignored by git.
Hprof files should also be ignored as those are only useful for debugging a Java process that crashed.

# How

I added them to the .gitignore

# Test Plan

N/A

Commit:
Add .settings and .project from VS Code to gitignore

ide commented:
What in VS Code is generating these files? Ex: VS Code supposedly stopped generating .settings three years ago: https://code.visualstudio.com/updates/v0_8_0

sergeichestakov commented:
Nevermind, they seem to have been generated by Buildship which is a gradle plugin for Eclipse. I don't have Eclipse installed so it must have been a gradle script somewhere.

![dotfiles](https://user-images.githubusercontent.com/24947334/56062446-41c14580-5d21-11e9-94b1-05a6c2bff8de.png)

Not sure if this has been a problem for anyone else? It might be good to add it just in case (esp if anyone uses Eclipse for whatever reason). Let me know what you think.

Commit:
Change title to Eclipse and add hprof files under Android

hprof is a tool built into JDK for profiling the CPU and heap usage within a JVM. This file may be produced when a Java process crashes and contains a heap dump of the process at the time of failure.
